Documentation/sourcebuild-in-repos/new-dependencies.md

```diff
@@ -0,0 +1,128 @@
+# Adding a new dependency
+
+## Basics
+
+When adding a new dependency, there are a few steps common between any type
+of dependency.  If you use darc, `darc add-dependency` will take care of
+this process for you.  Otherwise:
+
+1. Add the package and a default version to eng/Versions.props.  The default
+  version property should be named `<PackageName>PackageVersion`,
+  e.g. `SystemCollectionsImmutablePackageVersion`.
+1. Add the package and a repo reference to eng/Version.Details.xml.  This
+  should include a SourceBuild metadata entry, looking something like:
+
+```xml
+   <Dependency Name="Microsoft.Net.Compilers.Toolset"
+                  Version="4.2.0-3.22205.5"
+                  CoherentParentDependency="Microsoft.NET.Sdk">
+      <Uri>https://github.com/dotnet/roslyn</Uri>
+      <Sha>0167599e0e1634ea3ed8d0e41390a3c0d9b3e4e9</Sha>
+      <SourceBuild RepoName="roslyn" ManagedOnly="true" />
+    </Dependency>
+```
+
+1. Set up dependency flow from the foreign repo to your repo.
+
+In general, you should aim to use one version of each package.  If you are
+using a package as reference-only, it is possible to use multiple versions,
+but only one implementation version of each package will be used -
+source-build will override it to the version that is being built in this
+version of the SDK.
+
+The source-build metadata is important - this tells source-build which repo
+package contains the specific nupkg you want.
+
+Another uncommon case that can cause problems is when the repo `<Dependency>`
+version and the source-built intermediate package version don't match.
+In this case, a direct dependency should be added to the source-build
+intermediate package and the CoherentParentDependency attribute should be
+set to the repo that consumes the dependency. For an example,
+see [installer's F# dependency](https://github.com/dotnet/installer/blob/ba1739a2363b1062f03ea386ec67174c6468d3b2/eng/Version.Details.xml#L128).
+You can find the version needed by running `darc get-build`
+or using [BAR](https://aka.ms/bar).
+
+## Microsoft dependencies
```

Comment:
What do you think about combining the Microsoft and External sections?  The issue is there are Microsoft dependencies that don't meet any of the cases specified in this section and get pushed to source-build-externals.  Furthermore, we have treated some Microsoft dependencies as text-only in the past.

Reply:
That sounds fine to me, changed as suggested.